Tools/Scripts/webkitpy/benchmark_runner/data/patches/webserver/MotionMark1.3.patch

```diff
@@ -0,0 +1,142 @@
+-        if (!benchmarkController.options)
+-            return false;
+-
+-        benchmarkController.suites = suitesManager.suitesFromQueryString(benchmarkController.options["suite-name"], benchmarkController.options["test-name"]);
++        benchmarkController.suites = suitesManager.constructSuitesFromURLParameters(location.search)
+         if (!benchmarkController.suites.length)
+             return false;
+ 
+         setTimeout(function() {
+-            this._startBenchmark(benchmarkController.suites, benchmarkController.options, "running-test");
++            this._startBenchmark(benchmarkController.suites, this.benchmarkDefaultParameters, "running-test");
+         }.bind(this), 0);
+         return true;
+     },
+diff --git a/resources/runner/motionmark.js b/resources/runner/motionmark.js
+index a2ea11401414e03ea07ea94be8761f0542da515a..5f7f4f1d791c2392edf1df61f4d45665e4d63ced 100644
+--- a/resources/runner/motionmark.js
++++ b/resources/runner/motionmark.js
+@@ -420,6 +420,62 @@ window.benchmarkRunnerClient = {
+     didFinishLastIteration: function()
+     {
+         benchmarkController.showResults();
++    },
++
++    _computeTestReport: function(testName, testResults)
++    {
++        return [testResults[Strings.json.score]];
++    },
++
++    _computeSuiteReports: function(suiteName, suiteResults)
++    {
++        var testsReports = {};
++        for (var testName in suiteResults) {
++            var testReports = this._computeTestReport(testName, suiteResults[testName]);
++            testsReports[testName] = { "metrics": { "Score": { "current": testReports } } };
++        }
++        return testsReports;
++    },
++
++    _computeIterationReports: function(iterationResults)
++    {
++        var suitesReports = {};
++        var suffix = Strings.version ? '-' + Strings.version : '';
++        for (var suiteName in iterationResults[Strings.json.results.tests]) {
++            var testsReports = this._computeSuiteReports(suiteName, iterationResults[Strings.json.results.tests][suiteName]);
++            suitesReports[suiteName + suffix] = { "metrics": { "Score": ["Geometric"] }, "tests": testsReports };
++        }
++        return suitesReports;
++    },
++
++    _computeResultsReports: function()
++    {
++        var results = this.results.results[0];
++        var iterationReports = this._computeIterationReports(results);
++        iterationReports['debugOutput'] = {
++            options: benchmarkRunnerClient.results.options,
++            data: benchmarkRunnerClient.results.data,
++        }
++        return iterationReports;
++    },
++
++    didFinishLastIteration: function()
++    {
++       // Submit result to server
++       var results = JSON.stringify(this._computeResultsReports());
++       var xhr = new XMLHttpRequest();
++       xhr.open("POST", "/report");
++       xhr.setRequestHeader("Content-type", "application/json");
++       xhr.setRequestHeader("Content-length", results.length);
```

Comment:
Let's do `Buffer.from(results).length` instead so that UTF characters can be handled correctly.

Reply:
Does this exist in regular JavaScript?